internal/pkg/skuba/addons/cilium.go

```diff
@@ -55,26 +58,55 @@ func (renderContext renderContext) CiliumImage() string {
 }
 
 func renderCiliumTemplate(addonConfiguration AddonConfiguration) string {
-	return ciliumManifest
+	ciliumVersion := kubernetes.AddonVersionForClusterVersion(kubernetes.Cilium, addonConfiguration.ClusterVersion).Version
+	switch {
+	case strings.HasPrefix(ciliumVersion, "1.5"):
+		return ciliumManifestv15
+	case strings.HasPrefix(ciliumVersion, "1.6"):
+		return ciliumManifestv16
+	}
+	panic(fmt.Sprintf("invalid cilium addon version: %s", ciliumVersion))
+}
+
+func renderCiliumPreflightTemplate(addonConfiguration AddonConfiguration) string {
+	ciliumVersion := kubernetes.AddonVersionForClusterVersion(kubernetes.Cilium, addonConfiguration.ClusterVersion).Version
```

Comment:
I think there is a bug here. When trying the following update:
```
Current Kubernetes cluster version: 1.16.2
Latest Kubernetes version: 1.18.0
```
addonConfiguration.ClusterVersion={{1.16.2 1.16.1}

In kubernetes/versions.go maps to "Cilium:        &AddonVersion{"1.5.3", 2},"

Therefore, it never enters the switch in line 73 and returns "", so no preflight pod gets deployed and as a consequence:
```
E0421 16:53:31.390515    3164 addons.go:364] failed on "cilium" addon BeforeApply callback: could not find cilium pod
E0421 16:53:31.391062    3164 addons.go:170] failed to apply "cilium" addon (could not find cilium pod)
```
addonConfiguration.ClusterVersion should point to Latest Kubernetes version, not to the current one (maybe kubernetes.LatestVersion().String()?)

Reply:
This works:
`ciliumVersion := kubernetes.AddonVersionForClusterVersion(kubernetes.Cilium, kubernetes.LatestVersion()).Version`

Reply:
I think it's a correct behavior. If your k8s version is 1.16, Cilium 1.6 should **not** be deployed. We want to deploy Cilium 1.6 for k8s >= 1.17. You should deploy newer Kubernetes.

Reply:
What I want to say is that the logic checks the current K8s version instead of the target K8s version. In the upgrade action, I think it should check what is the target version and fetch Cilium version from there

Reply:
We upgrade only addons here, so we want to stay with the current k8s version. Changing k8s is something which should be done with `skuba cluster upgrade`.